client/src/js/services/SupplierService.js

```diff
@@ -0,0 +1,41 @@
+angular.module('bhima.services')
+.service('SupplierService', SupplierService);
+
+SupplierService.$inject = [ '$http', 'util' ];
+
+function SupplierService($http, util) {
+  var service = this;
+
+  service.create = create;
+  service.read = read;
+  service.update = update;
+  service.filter = filter; 
+
+
+  function create(supplier) {
+    return $http.post('/suppliers', supplier)
+    .then(util.unwrapHttpResponse);
+  }
+
+  function read(uuid, params) {
+     var url = (uuid) ? '/suppliers/' + uuid : '/suppliers';
```

Comment:
This syntax
```js
var url = '/suppliers/'.concat(uuid || '');
```
introduced by @DedrickEnc, is much more readable.  Could you use it instead?